images/cassandra-recovery/pkg/controller/controller.go

```diff
@@ -0,0 +1,151 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	uruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/mesosphere/kudo-cassandra-operator/images/cassandra-recovery/pkg/sts"
+)
+
+const (
+	InstanceLabel = "kudo.dev/instance"
+)
+
+type Controller struct {
+	client     *kubernetes.Clientset
+	queue      workqueue.RateLimitingInterface
+	informer   cache.SharedIndexInformer
+	maxRetries int
+}
+
+func NewController(client *kubernetes.Clientset) *Controller {
+	return &Controller{
+		client: client,
+	}
+}
+
+func (c *Controller) Run(ctx context.Context) {
+	namespace := os.Getenv("NAMESPACE")
```

Comment:
Let's decouple controller configuration from running the controller. How about retrieving the env vars in `main.go` and using them as arguments for `Run`?

```go
func (c *Controller) Run(ctx context.Context, instance string, namespace string)
```

Reply:
I've moved that into a separate Options